lib/legacy/zstd_legacy.h

```diff
@@ -28,6 +28,31 @@ extern "C" {
 #include "zstd_v06.h"
 #include "zstd_v07.h"
 
+#ifndef ZSTD_LEGACY_SUPPORT
```

Comment:
Another use case to properly support is if `ZSTD_LEGACY_SUPPORT` is defined with value `0`.
This is up to now supported, and means "no legacy support".

But in proposed code, it seems it would mean the otherwise, supporting _all_ legacy versions.